pwndbg/dbg/gdb/__init__.py

```diff
@@ -430,54 +430,70 @@ def vmmap(self) -> pwndbg.dbg_mod.MemoryMap:
         pages.sort()
         return GDBMemoryMap(qemu, pages)
 
+    def _find_memory_last_readable(self, start: int, count: int) -> int:
+        end = start + count
+        result = -1
+
+        def _is_readable(addr) -> bool:
+            try:
+                self.read_memory(addr, 1)
+                return True
+            except pwndbg.dbg_mod.Error:
+                return False
+
+        if not _is_readable(start):
+            return result
+
+        while start <= end:
+            mid = (start + end + 1) // 2
+            if _is_readable(mid):
+                result = mid
+                start = mid + 1
+            else:
+                end = mid - 1
+
+        return result
+
     @override
     def read_memory(self, address: int, size: int, partial: bool = False) -> bytearray:
-        result = b""
         count = max(int(size), 0)
         addr = address
 
         try:
             result = gdb.selected_inferior().read_memory(addr, count)
+            return bytearray(result)
         except gdb.error as e:
             if not partial:
                 raise pwndbg.dbg_mod.Error(e)
 
-            message = str(e)
-
-            stop_addr = addr
-            match = re.search(r"Memory at address (\w+) unavailable\.", message)
-            if match:
-                stop_addr = int(match.group(1), 0)
+            if pwndbg.aglib.remote.is_remote():
+                # GDB remote debugging will return the start address as the failed
```

Comment:
1) Lets first handle local debugging instead of remote first
2) Lets expand on the comment
3) Add link to the GDB bug if its already reported. Report if not.
4) Add a general comment to the except part such as:
> Memory read failed - likely bcoz part of the memory is unmapped


Then a comment for remote:
> On remote debugging, the error message contains start address instead of stop address. This is a GDB bug tracked by <LINK>
> To work around this, we perform a binary search to find the proper stop address that does not fail